### PR TITLE
Simplifying params + consolidating repo

### PR DESCRIPTION
**Pull request by b1n-ch1kn**
Changes:
- Removed requirement for `eufs_msgs` repo being downloaded. 
- `config`->`eufs_config`: more accurate package name.
- `eufs_config/pluginParams.yaml`: now stores all params ROS 2 gazebo plugin nodes that are specific for `eufs_sim` operation. These get updated in a runtime copy of the file with user specific options from the launcher window, rather than being passed through multiple nodes + launch files. Also global params are declared for all nodes, rather than duplicating.
- `eufs_config/launcherOptions.yaml`: updated names of user options to match ros params so they can be assigned easily.
- `*/helpers_gazebo`: each plugin now loads plugins from yaml file, not XML/SDF.
- `eufs_launcher/launch/simulation.launch.py` + `eufs_racecar/launch/load_car.launch.py`: combined these, seemed unnecessary to pass so many args for just two new launched nodes in `load_car`.
- `*.ui`: all `.ui` RQT files have been moved to a `ui/` directory in package, rather than `resource` which is occupied by the package index file.
- `eufs_rqt/MissionControlGUI`: previously deprecated with custom state control, finally removed node and with it, the `eufs_msgs` dependency.
- `eufs_racecar/robot.urdf.xacro`: merged ROS 2 plugins xacro into robot xacro.
- `eufs_tracks`: remove a bunch of legacy redundant files. Also moved template gazebo generation files to their specific `models` or `worlds` folder.

### CODE REVIEW

**Review by supex0fan — APPROVED** [review]

File: eufs_config/config/launcherOptions.yaml [hunk 1]
```diff
@@ -56,35 +56,35 @@ eufs_launcher:
       label: "Ground Truth TF"
       checked_on_default: Yes
       parameter_triggering:
-        if_on: { "publish_gt_tf:=true" }
-        if_off: { "publish_gt_tf:=false" }
+        if_on: { "publish_transform:=true" }
```
Comment (Grant-ed) [inline eufs_config/config/launcherOptions.yaml]: Should this not be renamed to publish_ground_truth_transform? publish_transform seems ambiguous if it is specifically referring to Ground Truth TF, especially as this PR is about simplifying params

File: eufs_launcher/ui/launcher.ui
Comment (Grant-ed) [inline eufs_launcher/ui/launcher.ui:1]: Is it worth renaming the "resource" folder's to "ui" across our packages? I get that it makes sense in the simplifying names case, but I feel like it's not exactly best practice straying from ros convention (from my understanding).

File: eufs_plugins/gazebo_lidar_plugin/include/gazebo_lidar_plugin/gazebo_ros_lidar.hpp [hunk 2]
```diff
@@ -69,20 +67,19 @@ class GazeboRosLidarPrivate;
...
+    virtual ~GazeboRosLidar();
 
-  // Documentation Inherited
-  void Load(gazebo::sensors::SensorPtr _parent, sdf::ElementPtr _sdf);
+    // Documentation Inherited
+    void Load(gazebo::sensors::SensorPtr _parent, sdf::ElementPtr _sdf);
 
-private:
-  std::unique_ptr<GazeboRosLidarPrivate> impl_;
+   private:
```
Comment (Grant-ed) [inline eufs_plugins/gazebo_lidar_plugin/include/gazebo_lidar_plugin/gazebo_ros_lidar.hpp:81]: I have no idea why pre-commit wants 3 spaces before "private:" and "public:", instead of 2 space indentation like the rest of the file - is this something that is intentional and/or desired? If not, can we fix it?

File: eufs_launcher/ui/launcher.ui
Comment (b1n-ch1kn) [reply eufs_launcher/ui/launcher.ui:1]: "no"
tldr, python ros needs the resource for indexing, and ros standard is to separate UI like you would Config or Rviz folders

File: eufs_plugins/gazebo_lidar_plugin/include/gazebo_lidar_plugin/gazebo_ros_lidar.hpp (hunk 2 above)
Comment (b1n-ch1kn) [reply eufs_plugins/gazebo_lidar_plugin/include/gazebo_lidar_plugin/gazebo_ros_lidar.hpp:81]: blame google :)
its fine i like it

Comment (Grant-ed) [reply eufs_plugins/gazebo_lidar_plugin/include/gazebo_lidar_plugin/gazebo_ros_lidar.hpp:81]: Thanks I Hate It
